### PR TITLE
[Microsoft Windows Defender ATP] Update plugin to be cloud enabled

### PR DESCRIPTION
**Pull request by rbowden-r7**
### Description

Describe the proposed changes:

  - to update the microsoft atp plugin to be cloud enabled 
  - to fix a bug where using machine_id as a praameter can caused an error, due to the look up on the id

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Testing

All testing has been added to this wiki page for the testing all of the actions for 

4.8.1 orchestrator 

5.0.0 orchestrator

5.0.0 cloud 

https://wiki.corp.rapid7.com/display/KOMAND/Microsoft+Windows+Defender+ATP+Plugin+testing

Due to staging server issues testing for triggers were done locally 

https://issues.corp.rapid7.com/browse/PLGN-387?focusedCommentId=2106465&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-2106465

edit tests have now been carried out on staging as well and are working as expected 

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

**Review by igorski-r7 — COMMENTED** [review]
Please also take a look at the actions that got huge update to `schema.py` file if they work fine 😄 New tooling likes to break some of the json validation there.

File: plugins/microsoft_atp/Dockerfile [hunk 1]
```diff
@@ -1,36 +1,20 @@
-FROM rapid7/insightconnect-python-3-38-slim-plugin:4
-# Refer to the following documentation for available SDK parent images: https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide
+FROM rapid7/insightconnect-python-3-38-plugin:4
```
[the remark this answers is not in the document]
Comment (rbowden-r7) [reply plugins/microsoft_atp/Dockerfile:1]: does `FROM rapid7/insightconnect-python-3-38-plugin:4` need updated or do I just need to add in the plugin spec file ?

File: plugins/microsoft_atp/help.md [hunk 2]
```diff
@@ -50,229 +40,232 @@ Example input:
...
+    "MITRE Techniques": [
+      {}
+    ],
+    "RBAC Group IDs": {},
+    "RBAC Group Names": {},
+    "Recommended Actions": {},
+    "Severity": {},
+    "Source": {},
+    "Source Type": {},
+    "Title": {}
```
[the remark this answers is not in the document]
Comment (rbowden-r7) [reply plugins/microsoft_atp/help.md]: I have added examples to all actins and triggers

File: plugins/microsoft_atp/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (igorski-r7) [reply plugins/microsoft_atp/Dockerfile:1]: Please add the following 

```
sdk:
  type: full
  version: 5
  user: nobody
```

to the `plugin.spec.yaml` file 😄 The refresh the plugin. It should automatically update the SDK in the Dockerfile

**Review by igorski-r7 — COMMENTED** [review]
Please refresh the plugin to update the `Dockerfile` with `insight-plugin refresh` 😄

**Comment by rbowden-r7** [comment]
> Please refresh the plugin to update the `Dockerfile` with `insight-plugin refresh` 😄

The docker file has been refreshed and the new version of the sdk is now added

File: plugins/microsoft_atp/Dockerfile [hunk 3]
```diff
@@ -1,36 +1,20 @@
...
-
-# Need gcc for maya
-RUN apk add --no-cache --virtual .build-deps \
-                                        make \
-                                        gcc \
-                                        libc-dev \
-                                        libffi-dev \
-                                        openssl-dev \
-                                        libxml2-dev \
-                                        libxslt-dev
```
Comment (igorski-r7) [inline plugins/microsoft_atp/Dockerfile:18]: Is that necessary to be removed? If its required for a plugin to run try add the following to the `sdk` in `plugin.spec.yaml`

```
sdk:
  ...
  packages:
    - make
    - gcc
    etc...
```

Comment (rbowden-r7) [reply plugins/microsoft_atp/Dockerfile:18]: This was removed when the refresh tool was run, however when I make the new docker image using this docker file and exec the container we can see that that maya is still installed and that the actions work as expected

```
docker exec -it atp /bin/sh

$ pip list | grep maya
WARNING: The directory '/nonexistent/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
maya                          0.5.0
```

Comment (igorski-r7) [reply plugins/microsoft_atp/Dockerfile:18]: Thats fine 😄

**Review by igorski-r7 — APPROVED** [review]

**Review by ablakley-r7 — APPROVED** [review]

**Pull request merged**